test.mk

```diff
@@ -161,50 +161,255 @@ spec:
+    - "http"
+    - "https"
+    paths:
+      "/**":
+        get:
+          operationId: Get
+          responses:
+            '200':
+              description: Get
+            default:
+              description: Error
+        post:
+          operationId: Post
+          responses:
+            '200':
+              description: Post
+            default:
+              description: Error
+        put:
+          operationId: Put
+          responses:
+            '200':
+              description: Put
+            default:
+              description: Error
+    security:
+    - google_jwt: []
+    securityDefinitions:
+      google_jwt:
+        authorizationUrl: ""
+        flow: "implicit"
+        type: "oauth2"
+        x-google-issuer: "https://cloud.google.com/iap"
+        x-google-jwks_uri: "https://www.gstatic.com/iap/verify/public_key-jwk"
+        x-google-audiences: "{{JWT_AUDIENCE}}"
+    host: "{{HOSTNAME}}"
+    x-google-endpoints:
+    - name: "{{HOSTNAME}}"
+      target: "{{IP_ADDRESS}}"
+endef 
 
 export TEST_APP
 service%-deployment.yaml:
 	@echo "$${TEST_APP}" | \
-	  sed -e "s/{{NAME}}/service$*/g" \
+	 sed -e "s/{{NAME}}/service$*/g" \
 	> $@
 
 export TEST_SERVICE_INGRESS
 service%-simple-ingress-cloudep.yaml: project
 	@echo "$${TEST_SERVICE_INGRESS}" | \
-	  sed -e "s/{{NAME}}/service$*/g" \
+	 sed -e "s/{{NAME}}/service$*/g" \
 		-e "s/{{PROJECT}}/$(PROJECT)/g" \
-	    -e "s/{{INGRESS_NAME}}/service$*-ingress/g" \
-	  > $@
+	 -e "s/{{INGRESS_NAME}}/service$*-ingress/g" \
+	 > $@
 
 export TEST_INGRESS
 service%-ingress.yaml: project
 	@echo "$${TEST_INGRESS}" | \
-	  sed -e "s/{{NAME}}/service$*/g" \
+	 sed -e "s/{{NAME}}/service$*/g" \
 		-e "s/{{PROJECT}}/$(PROJECT)/g" \
-	  > $@
+	 > $@
 
 export TEST_SERVICE_SIMPLE
 service%-simple-cloudep.yaml: project
 	@echo "$${TEST_SERVICE_SIMPLE}" | \
-	  sed -e "s/{{NAME}}/service$*/g" \
+	 sed -e "s/{{NAME}}/service$*/g" \
 		-e "s/{{PROJECT}}/$(PROJECT)/g" \
-	    -e "s/{{IP_ADDRESS}}/0\.0\.0\.0/g" \
+	 -e "s/{{IP_ADDRESS}}/0\.0\.0\.0/g" \
 	  > $@
 
 export TEST_SERVICE
 service%-cloudep.yaml: project
 	@echo "$${TEST_SERVICE}" | \
-	  sed -e "s/{{NAME}}/service$*/g" \
-	    -e "s/{{HOSTNAME}}/service$*.endpoints.$(PROJECT).cloud.goog/g" \
-	    -e "s/{{IP_ADDRESS}}/0\.0\.0\.0/g" \
+	 sed -e "s/{{NAME}}/service$*/g" \
+		-e 	"s/{{TITLE}}/Test Spec/g"\
+		-e 	"s/{{PROJECT}}/$(PROJECT)/g"\
+	 -e "s/{{HOSTNAME}}/service$*.endpoints.$(PROJECT).cloud.goog/g" \
+	 -e "s/{{IP_ADDRESS}}/0\.0\.0\.0/g" \
 		-e "s/{{JWT_AUDIENCE}}//g" \
 	  > $@
 
+export TEST_SERVICE_SPEC_INGRESS_CLOUDEP
+service%-cloudep-template-spec-ing.yaml: project
+	@echo "$${TEST_SERVICE_SPEC_INGRESS_CLOUDEP}" | \
+		sed -e "s/{{NAME}}/service$*/g" \
+		-e 	"s/{{TITLE}}/Test Template Spec/g"\
+		-e 	"s/{{PROJECT}}/$(PROJECT)/g"\
+	 -e "s/{{HOSTNAME}}/{{.Endpoint}}/g" \
+	 -e "s/{{IP_ADDRESS}}/{{.Target}}/g"  \
+		-e "s/{{INGRESS_NAME}}/service$*-ingress/g" \
+  -e "s/{{JWT_AUDIENCE}}//g" \
```

Comment:
Do not delete the JWTAudience per prior comments.

Reply:
This kind of templating will require us to turn the openAPISpec: field from a map into a string which makes sense to me, but it is a breaking change for existing uses.